[TASK] Setup Canada DC API call

mariasenosiain opened the pull request:
By submitting a PR to this repository, you agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md). Please see the [contributing guidelines](https://github.com/auth0/.github/blob/master/CONTRIBUTING.md) for how to create and submit a high-quality PR for this repo.

### Description

Provides public API method to use Canada DC: `YotiDocScan.useCanadaService();
This is intentionally not documented because this method is provided for a specific YDS client. 
**If you're not a Yoti member and you see this PR, please, do not use this method unless you're authorised to do so.**

### References

Android client update: [JIRA](https://lampkicking.atlassian.net/browse/YD-2747)
iOS client update: [JIRA](https://lampkicking.atlassian.net/browse/YD-2746)

### Testing
- Create session in Canada DC production and use sessionId and token provided for the following checks:
- Add this configuration call inside App.js file, before `YotiDocScan.startSession(sessionId, clientSessionToken, onSuccess, onError);`
```
YotiDocScan.useCanadaService();
```
- Inside example folder, perform yarn install (remove previous node_modules folder):
- [x] Android client performs Canada DC call:
1. Start Android client via `yarn run android`.
2. Introduce Canada sessionId & session token.
3. Session is started.

- [x] iOS client performs Canada DC call
1. Inside example folder, go to ios folder and execute `pod install`
2. Start iOS client via `yarn run ios`.
2. Introduce Canada sessionId & session token.
3. Session is started.

### Checklist

- [x] I have added documentation for new/changed functionality in this PR or in auth0.com/docs
- [x] All active GitHub checks for tests, formatting, and security are passing
- [x] The correct base branch is being used, if not `master`

mariasenosiain committed:
Setup Android/iOS exported function to configure Canada DC

Related to YD-2747

mariasenosiain commented on ios/RNYotiDocScanManager.m:
```diff
@@ -57,6 +63,14 @@ - (NSString * _Nonnull)sessionTokenFor:(YotiSDKNavigationController * _Nonnull)n
     return self.sessionToken;
 }
 
+- (ServerLocation)serverLocationFor:(YotiSDKNavigationController * _Nonnull)navigationController {
```
Hi @asklausen !
As we were talking about this morning, I'm a little bit sticky with this config...could you help me with it pls? I'm not sure how I have to manage an enum defined in Swift from Obj-C 😅  And I'm pretty sure you're going to smash it in 1 min :happy:
Thanks!

asklausen commented on ios/RNYotiDocScanManager.m:
```diff
@@ -57,6 +62,14 @@ - (NSString * _Nonnull)sessionTokenFor:(YotiSDKNavigationController * _Nonnull)n
 }
 
+- (ServerLocation)serverLocationFor:(YotiSDKNavigationController * _Nonnull)navigationController {
+    if (self.setupCanadaDataCenter) {
+        return ServerLocation.canada;
+    } else {
+        return ServerLocation.unitedKingdom;
+    }       
+ }
```
```suggestion
    if (self.setupCanadaDataCenter) {
        return ServerLocationCanada;
    } else {
        return ServerLocationUnitedKingdom;
    }       
 }
```

mariasenosiain replied:
🤦🏻‍♀️ Thanks!

asklausen commented on ios/RNYotiDocScanManager.m:
```diff
@@ -14,6 +15,7 @@ @interface RNYotiDocScanManager ()
 @property (nonatomic, strong) UIViewController *rootViewController;
 @property (nonatomic, strong) NSString *sessionID;
 @property (nonatomic, strong) NSString *sessionToken;
+@property (nonatomic) BOOL setupCanadaDataCenter;
```
In this case, I think it's more correct to say `setUp` (would have been lowecased if used in the context of e.g. `performSetup`)
```suggestion
@property (nonatomic, assign) BOOL setUpCanadaDataCenter;
```

mariasenosiain replied:
:indeed:

mariasenosiain committed:
Bump up version to 1.6.0

mariasenosiain committed:
Merge branch 'release/1.6.0' into YD-2747

mariasenosiain committed:
Fix warning issue on iOS wrapper

Custom init was producing warnings related with iOS threading management
Back to boolean flag to set up the Canada Data center

Related to YD-2747

Zzhelyaz commented:
Verified.
